prow/gerrit/client/client_test.go

```diff
@@ -845,6 +868,48 @@ func TestQueryChange(t *testing.T) {
 			},
 			revisions: map[string][]string{},
 		},
+		{
+			name: "one up-to-date change found twice due to pagination. Duplicate should be removed",
+			lastUpdate: map[string]time.Time{
+				"bar": now.Add(-time.Hour),
+			},
+			changes: map[string][]gerrit.ChangeInfo{
+				"foo": {
+					{
+						Project:         "bar",
+						ID:              "1",
+						Number:          1,
+						CurrentRevision: "1-1",
+						Updated:         makeStamp(now.Add(-time.Minute)),
+						Revisions: map[string]gerrit.RevisionInfo{
+							"1-1": {
+								Created: makeStamp(now.Add(-time.Minute)),
+							},
+						},
+						Status: "NEW",
+					},
+					{
+						Project:         "bar",
+						ID:              "1",
+						Number:          1,
+						CurrentRevision: "1-2",
+						Updated:         makeStamp(now),
+						Revisions: map[string]gerrit.RevisionInfo{
+							"1-1": {
+								Created: makeStamp(now.Add(-time.Minute)),
+							},
+							"1-2": {
+								Created: makeStamp(now),
+							},
+						},
+						Status: "NEW",
+					},
```

Comment:
nit: This unit test didn't catch the bug because it doesn't include enough changes to exercise pagination. We could add more changes to the end of this slice to validate that case.

Reply:
Does the test seem satisfactory now?

Reply:
We need more results than the pagination limit. This example would be fine if we use a value of `2`  instead of `5` for the pagination limit on line 1079.